v2/ical.NET.UnitTests/DataTypes/RequestStatusTest.cs

```diff
@@ -1,123 +1,127 @@
-﻿using Ical.Net.DataTypes;
+﻿using System.Collections.Generic;
+using Ical.Net.DataTypes;
 using Ical.Net.Interfaces.General;
 using NUnit.Framework;
+using NUnit.Framework.Interfaces;
 
 namespace Ical.Net.UnitTests
 {
-    [TestFixture]
     public class RequestStatusTest
     {
-        [Test]
-        public void InitialObjectState_Constructor_Default()
+        private static IEnumerable<ITestCaseData> ConstructorTestCases() 
         {
-            var requestStatus = new RequestStatus();
+            var emptyRequestStatus = new RequestStatus();
+            yield return new TestCaseData(emptyRequestStatus, null, null, null)
+                .SetName("Default request status constructor has null properties");
 
-            Assert.IsNull(requestStatus.StatusCode);
-            Assert.IsNull(requestStatus.Description);
-            Assert.IsNull(requestStatus.ExtraData);
+            var invalidSerializedValue = new RequestStatus("ABC");
+            yield return new TestCaseData(invalidSerializedValue, null, null, null)
+                .SetName("Constructor with invalid serialised text has null properties");
         }
 
-        [Test]
-        public void InitialObjectState_Constructor_InvalidSerialisedValue() 
+        [Test, TestCaseSource(nameof(ConstructorTestCases))]
+        public void ConstructorTests(RequestStatus incoming, string expectedDescription, string expectedStatusCode, string expectedExtraData) 
         {
-            var requestStatus = new RequestStatus("ABC");
-
-            Assert.IsNull(requestStatus.StatusCode);
-            Assert.IsNull(requestStatus.Description);
-            Assert.IsNull(requestStatus.ExtraData);
+            Assert.AreEqual(incoming.StatusCode, expectedStatusCode);
+            Assert.AreEqual(incoming.Description, expectedDescription);
+            Assert.AreEqual(incoming.ExtraData, expectedExtraData);
         }
 
         [Test, Ignore("Understand what 'valid' values RequestStatusSerializer expects")]
         public void InitialObjectState_Constructor_ValidSerialisedValue() { }
 
-        [Test]
-        public void ToString_ValidInputWithNullStatusCode()
+        private static IEnumerable<ITestCaseData> ToStringTestCases() 
         {
-            var requestStatusA = new RequestStatus();
-            var requestStatusB = new RequestStatus()
+            var emptyRequestStatus = new RequestStatus();
+            yield return new TestCaseData(emptyRequestStatus, ";")
+                .SetName("ToString() for empty request status should return semicolon");
```

Comment:
I think you've written unit tests to assert the _current_ behavior instead of the _correct_ behavior. Why would an empty `RequestStatus.ToString()` return anything other than null or `string.Empty`?

I think this test should be assert that an empty request status returns `null`. I know it won't pass, so let's mark it `Ignore`d.